src/main/java/io/eurora/euroratest/archive/ArchiveService.java

```diff
@@ -0,0 +1,76 @@
+package io.eurora.euroratest.archive;
+
+import io.eurora.euroratest.EuroraTestConfiguration;
+import io.eurora.euroratest.EuroraTestException;
+import io.eurora.euroratest.PathService;
+import io.eurora.euroratest.report.ReportService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+import java.util.zip.ZipOutputStream;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class ArchiveService {
+
+  private final ZipService zipService;
+  private final UnzipService unzipService;
+  private final FilePool filePool;
+  private final ExecutorService taskExecutor;
```

Comment:
You need to do it so that the ExecutorService always returns a new instance. Because the thread pool is emptied after the call execute() method.
ExecutorService is now identified as a singleton bean.
We can define  ExecutorService as prototype bean in EuroraTestConfiguration and call it as a factory method.
private final ObjectProvider<ExecutorService> taskExecutorFactory;
var taskExecutor = taskExecutorFactory.getObject()

```suggestion
  private final ObjectProvider<ExecutorService> taskExecutorFactory;
  
  var taskExecutor = taskExecutorFactory.getObject()